addons/spreadsheet/static/src/chart/odoo_chart/odoo_bar_chart.js

```diff
@@ -58,6 +68,35 @@ function createOdooChartRuntime(chart, getters) {
             backgroundColor: color,
         };
         chartJsConfig.data.datasets.push(dataset);
+
+        const trend = chart.getDefinition().trend;
+        if (!trend?.display || chart.horizontal) {
+            continue;
+        }
+
+        const trendDataset = getTrendDatasetForBarChart(trend, dataset);
+        if (trendDataset) {
+            trendDatasets.push(trendDataset);
+        }
+    }
+
+    if (trendDatasets.length) {
+        /* We add a second x axis here to draw the trend lines, with the labels length being
+         * set so that the second axis points match the classical x axis
+         */
+        const maxLength = Math.max(
+            ...trendDatasets.map((trendDataset) => trendDataset.data.length)
+        );
+        chartJsConfig.options.scales[TREND_LINE_XAXIS_ID] = {
+            ...chartJsConfig.options.scales.x,
+            labels: Array(maxLength).fill(""),
+            offset: false,
+            display: false,
+        };
+        /* These datasets must be inserted after the original
+         * datasets to ensure the way we distinguish the originals and trendLine datasets after
+         */
+        trendDatasets.forEach((x) => chartJsConfig.data.datasets.push(x));
```

Comment:
you should add the equivalent of this commit -  https://github.com/odoo/o-spreadsheet/pull/4803/commits/14f40f43166c055d4d7bfa60a438301f51928375